ext/soap/php_http.c

```diff
@@ -241,6 +241,13 @@ static php_stream* http_connect(zval* this_ptr, php_url *phpurl, int use_ssl, ph
 	if (stream && *use_proxy && use_ssl) {
 		smart_str soap_headers = {0};
 
+		/* Set peer_name or name verification will try to use the proxy server name */
+		if (context && (tmp = php_stream_context_get_option(context, "ssl", "peer_name")) != NULL) {
```

Comment:
This condition seems to be the exact opposite of what the corresponding http_fopen_wrapper code does: https://github.com/php/php-src/blob/master/ext/standard/http_fopen_wrapper.c#L241 The http_fopen_wrapper variant looks right to me in that it still allows a manual peer_name override.